capa/features/extractors/cape/extractor.py

```diff
@@ -54,44 +56,43 @@ def extract_global_features(self) -> Iterator[Tuple[Feature, Address]]:
 
     def get_processes(self) -> Iterator[ProcessHandle]:
-        yield from capa.features.extractors.cape.file.get_processes(self.behavior)
+        yield from capa.features.extractors.cape.file.get_processes(self.report)
 
     def extract_process_features(self, ph: ProcessHandle) -> Iterator[Tuple[Feature, Address]]:
-        yield from capa.features.extractors.cape.process.extract_features(self.behavior, ph)
+        yield from capa.features.extractors.cape.process.extract_features(ph)
 
     def get_threads(self, ph: ProcessHandle) -> Iterator[ThreadHandle]:
-        yield from capa.features.extractors.cape.process.get_threads(self.behavior, ph)
+        yield from capa.features.extractors.cape.process.get_threads(ph)
 
     def extract_thread_features(self, ph: ProcessHandle, th: ThreadHandle) -> Iterator[Tuple[Feature, Address]]:
-        yield from capa.features.extractors.cape.thread.extract_features(self.behavior, ph, th)
+        if False:
+            # force this routine to be a generator,
+            # but we don't actually have any elements to generate.
+            yield Characteristic("never"), NO_ADDRESS
+        return
 
     def get_calls(self, ph: ProcessHandle, th: ThreadHandle) -> Iterator[CallHandle]:
-        yield from capa.features.extractors.cape.thread.get_calls(self.behavior, ph, th)
+        yield from capa.features.extractors.cape.thread.get_calls(ph, th)
 
     def extract_call_features(
         self, ph: ProcessHandle, th: ThreadHandle, ch: CallHandle
     ) -> Iterator[Tuple[Feature, Address]]:
-        yield from capa.features.extractors.cape.call.extract_features(self.behavior, ph, th, ch)
+        yield from capa.features.extractors.cape.call.extract_features(ph, th, ch)
 
     @classmethod
     def from_report(cls, report: Dict) -> "CapeExtractor":
-        cape_version = report["info"]["version"]
-        if cape_version not in TESTED_VERSIONS:
-            logger.warning("CAPE version '%s' not tested/supported yet", cape_version)
-
-        static = report["static"]
-        format_ = list(static.keys())[0]
-        static = static[format_]
-        static.update(report["behavior"].pop("summary"))
-        static.update(report["target"])
-        static.update({"processtree": report["behavior"]["processtree"]})
-        static.update({"strings": report["strings"]})
-        static.update({"format": format_})
-
-        behavior = report.pop("behavior")
-        behavior["network"] = report.pop("network")
-
-        return cls(cape_version, static, behavior)
+        cr = CapeReport.model_validate(report)
+
+        if cr.info.version not in TESTED_VERSIONS:
+            logger.warning("CAPE version '%s' not tested/supported yet", cr.info.version)
+
+        if cr.static is None:
+            raise UnsupportedFormatError("CAPE report missing static analysis")
+
+        if cr.static.pe is None:
+            raise UnsupportedFormatError("CAPE report missing PE analysis")
```

Comment:
does CAPE support other file formats? non-PE files?

Reply:
Yes, see https://capev2.readthedocs.io/en/latest/usage/packages.html

Reply:
good call. will need to find some examples of these other reports. i'll scrape through the Avast dataset first.